### DISCUSSION TITLE
Please delete

### DISCUSSION THREAD

**Discussion by Sifxx** [etsy/open-api#1003] [Q&A] [answered]
My listings have seven images and I am trying to overwrite the 4th image. But it doesn't matter how I structure my code it always places the new image as my primary image. The python function I have is 

```
def update_listing_image(listing_id, access_token, image_id, shop_id):    
    url = f"https://openapi.etsy.com/v3/application/shops/{shop_id}/listings/{listing_id}/images"
    image_path = PATH_TO_FILE
    with open(image_path, "rb") as f:
        image_data = f.read()
        files = {
            "rank": 4,
            "overwrite": True,
            "name": FILE_NAME,
            "image": (FILE_NAME, image_data, 'image/png'),
        }
        headers = {
            "x-api-key": etsy_keystring,
            "Authorization": f"Bearer {access_token}"
        }
        response = requests.post(url, files=files, headers=headers)
        print(response)

        if response.status_code != 201:
            print(f"Error uploading image {image_path}: {response.text}")        
```
Am I doing something wrong with the way I am sending my request? It uploads the image fine it just won't overwrite or for the image to switch a new rank

**Comment by Sifxx** [comment, marked as answer]
I have fixed it myself by using  
`data ={
            "rank": 4,
            "overwrite": True,
}` 
and passing it into the post request

**Comment by Gareth064** [comment]
@Sifxx can you put the title of this discussion back to what it was, its fine that you solved it yourself, but with the discussion being here for others to see, it will be useful when someone else comes across a similar issue.
